grails-app/controllers/org/pih/warehouse/api/CombinedShipmentItemApiController.groovy

```diff
@@ -24,6 +24,21 @@ class CombinedShipmentItemApiController {
 
     def orderService
 
+    def getProductsInOrders = {
+        def minLength = grailsApplication.config.openboxes.typeahead.minLength
+        def vendor = Location.get(params.vendor)
+        def destination = Location.get(params.destination)
+        String[] terms = params?.name?.split(",| ")?.findAll { it }
+        if (params.name && params.name.size() < minLength) {
+            render([data: []])
+            return
+        }
+        def orderItems = orderService.getProductsInOrders(terms, destination, vendor)
+        def products = orderItems*.findAll{ it.orderItemStatusCode != OrderItemStatusCode.CANCELED && it.getQuantityRemainingToShip() > 0 }.flatten().toArray().collect { it.product }.unique()
```

Comment:
Why do you need flatten().toArray()?

Reply:
@jmiranda Because it was returning a list of lists without it.